lib/features/favourite/domain/notifiers/favourite_movies_notifier.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:loggy/loggy.dart';
+import 'package:movie_app/features/favourite/data/repositories/database_repository.dart';
+import 'package:movie_app/features/popular/domain/entities/movie.dart';
+import 'package:q_architecture/q_architecture.dart';
+
+final favouriteMoviesNotifierProvider =
+    NotifierProvider<FavouriteMoviesNotifier, Map<int, Movie>>(
+  FavouriteMoviesNotifier.new,
+  name: 'Favourite movies notifier provider',
+);
+
+class FavouriteMoviesNotifier extends SimpleNotifier<Map<int, Movie>> {
+  late FavouriteMoviesRepository _favouriteMoviesRepository;
+
+  @override
+  Map<int, Movie> prepareForBuild() {
+    _favouriteMoviesRepository = ref.watch(favouriteMoviesRepositoryProvider);
+    loadFavourites();
+    return {};
+  }
+
+  Future<void> toggleFavourite(Movie movie) async {
+    throttle(
+      () async {
+        if (state.containsKey(movie.id)) {
+          await _favouriteMoviesRepository.unfavouriteMovie(movie);
```

Comment:
Sa svim pozivima prema repozitorijima bi trebalo provjeriti rezultat je li uspjelo izvršavanje, pa ako nije ispisati grešku u konzolu ili još bolje kao toast poruku (global failure ili nešto tog tipa).